pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -674,3 +690,181 @@ func (c *APIClient) GetARandomNodeName(ctx context.Context) (string, error) {
+func decodeRelease(data string) []byte {
+	// base64 decode string
+	b, err := b64.DecodeString(data)
+	if err != nil {
+		return nil
+	}
+
+	// For backwards compatibility with releases that were stored before
+	// compression was introduced we skip decompression if the
+	// gzip magic header is not found
+	if bytes.Equal(b[0:3], magicGzip) {
+		r, err := gzip.NewReader(bytes.NewReader(b))
+		if err != nil {
+			return nil
+		}
+		defer r.Close()
+		b2, err := io.ReadAll(r)
+		if err != nil {
+			return nil
+		}
+		b = b2
+	}
+	return b
+}
+
+// getConfigmapList retrieves a list of configmaps for a given namespace
+func getConfigmapList(cl *APIClient, namespace string) ([]v1.ConfigMap, error) {
+	configmaps, err := cl.Cl.CoreV1().ConfigMaps(namespace).List(context.TODO(), metav1.ListOptions{TimeoutSeconds: &cl.timeoutSeconds})
+	if err != nil {
+		log.Errorf("Can't list configmaps from the API server: %s", err.Error())
+		return nil, err
+	}
+	return configmaps.Items, nil
+}
+
+// getSecretList retrieves a list of secrets for a given namespace
+func getSecretList(cl *APIClient, namespace string) ([]v1.Secret, error) {
+	secrets, err := cl.Cl.CoreV1().Secrets(namespace).List(context.TODO(), metav1.ListOptions{TimeoutSeconds: &cl.timeoutSeconds})
+	if err != nil {
+		log.Errorf("Can't list secrets from the API server: %s", err.Error())
+		return nil, err
+	}
+	return secrets.Items, nil
+}
+
+// Retrieve Helm chart user values from the API server for a given name and namespace looking through Configmaps
+func GetDatadogHelmUserValuesFromConfigmap(cl *APIClient, name string, namespace string) ([]byte, error) {
+	var b bytes.Buffer
+	var userConfig chartUserValues
+	var selector labels.Selector
+	var helmConfigmap *v1.ConfigMap
+
+	// Only a single release for a given name can be deployed at one time.
+	// Other releases present a different status.
+	selector = labels.Set{
+		"owner":  "helm",
+		"status": "deployed",
+		"name":   name,
+	}.AsSelector()
+
+	configmaps, err := getConfigmapList(cl, namespace)
+	if err != nil {
+		log.Errorf("Can't list configmaps from the API server: %s", err.Error())
+	}
+
+	for _, cm := range configmaps {
+		if selector.Matches(labels.Set(cm.ObjectMeta.Labels)) {
+			helmConfigmap = &cm
+			// Chart data is stored in a gzipped base 64 encoded string
+			b64EncodedGzipRelease := helmConfigmap.Data["release"]
+			decodedrelease := decodeRelease(b64EncodedGzipRelease)
+			err := json.Unmarshal(decodedrelease, &userConfig)
+			if err != nil {
+				log.Debugf("Unable to retrieve the config data: %s", err.Error())
+			}
+			configjson, err := json.Marshal(userConfig)
+			if err != nil {
+				log.Debugf("Can't marshall user values into a proper JSON: %s", err.Error())
+			}
+			configyaml, err := yaml.JSONToYAML(configjson)
+			if err != nil {
+				log.Debugf("Can't convert user values to YAML: %s", err.Error())
+			}
+			fmt.Fprintln(&b, string(configyaml))
+			return b.Bytes(), nil
```

Comment:
imo this logic can be shared with the `secret` equivalent function

Reply:
logic was moved to `decodeChartValuesFromRelease`